koerce/patterns.py

```diff
@@ -488,9 +488,9 @@ def match(self, value, ctx: Context):
 @cython.final
 @cython.cclass
 class InstanceOf(Pattern):
-    type_: type
+    type_: Any
```

Comment:
The generated code has changed from

```c
+0508:         if isinstance(value, self.type_):
  __pyx_t_1 = ((PyObject *)__pyx_v_self->type_);
  __Pyx_INCREF(__pyx_t_1);
  __pyx_t_2 = __Pyx_TypeCheck(__pyx_v_value, __pyx_t_1); 
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;
  if (likely(__pyx_t_2)) {
```

to 

```c
+0508:         if isinstance(value, self.type_):
  __pyx_t_1 = __pyx_v_self->type_;
  __Pyx_INCREF(__pyx_t_1);
  __pyx_t_2 = PyObject_IsInstance(__pyx_v_value, __pyx_t_1); if (unlikely(__pyx_t_2 == ((int)-1))) __PYX_ERR(0, 508, __pyx_L1_error)
  __Pyx_DECREF(__pyx_t_1); __pyx_t_1 = 0;
  if (likely(__pyx_t_2)) {
```

According to my local benchmarks it doesn't really impact the performance, so it should be good to go. Could you please add a comment mentioning this?

Reply:
Also could you please add a test case testing `InstanceOf(some_abc_base_class)` since the ABCs implement `__instancecheck__()`?

Reply:
Both done.